doc/modules.md

```diff
@@ -0,0 +1,88 @@
+Modularization
+============
+
+This document describes the changes for a more fine grained modularization
+of common libraries, e.g., Lunchbox
+
+## Requirements
+
+* Per-functionality targets
+* Improved build times for super projects
+
+## Implementation
+
+* All sub project targets are excluded from the default build, and
+  required targets are built only based on the dependencies.
+* Sub-libraries have a top-level header, e.g., lunchbox/log.h, and a
+  sub-directory with the implementation and CMakeLists, e.g.,
+  lunchbox/log/log.h and .cpp. The new common_add_library() cmake module
+  auto-generates the top-level header. New sub libraries set
+  NAME_OMIT_LIBRARY_HEADER.
+* The old common_library is deprecated, and stays during the transition
+  period.
+* Optional sub-targets are to be avoided and hidden by facades (esp. in
+  Lunchbox, see PersistentMap). If they are necessary, e.g, in hwsd,
+  downstream code needs to test for the target's availability (as done
+  e.g. in eq/server).
+* Servus will eventually be reintegrated as a lunchbox sub library.
```

Comment:
Freedom is Slavery!! We've always been happy with Lunchbox::Servus!!

(not sure how to read your comment, so going full Orwell here)